### PR TITLE
lib+routes: Add backup

### PR DESCRIPTION
**Pull request by nguyenhuedang**

### CODE REVIEW

File: lib/key.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const { readFile } = require("fs");
+const crypto = require("crypto");
```
Comment (khaq) [inline lib/key.js]: May use bluebird/promise promisify

File: lib/generate.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const crypto = require("crypto");
+const fs = require("fs");
+
+const { publicKey, privateKey } = crypto.generateKeyPairSync("rsa", {
```
Comment (khaq) [inline lib/generate.js]: single quote vs double quote

File: lib/test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const crypto = require('crypto');
+const { loadPublicKey, loadPrivateKey, encrypt, decrypt } = require('./key');
+const assert = require('assert');
+
+const ALGORITHM = 'aes256';
+const KEY_SIZE = 256;
```
Comment (khaq) [inline lib/test.js]: Size should in byte

File: routes/backup.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+const encryptionPublicKey = crypto.createPublicKey(process.env.ENCRYPTION_PUBLIC_KEY);
+
+const router = new Router();
+router.get('/', asyncHandler(async function (req, res) {
+  const key = crypto.randomBytes(KEY_SIZE / 8);
+  const initialVector = crypto.randomBytes(IV_SIZE / 8);
+  const cipher = crypto.createCipheriv(CIPHER_ALGORITHM, key, initialVector);
+  const encryptedKey = encrypt(encryptionPublicKey, key);
+  const encryptedShare = cipher.update(req.key.share, 'hex', 'hex') + cipher.final('hex');
```
Comment (khaq) [inline routes/backup.js:19]: 2 'hex'

Comment (nguyenhuedang) [reply routes/backup.js:19]: I use this signature

```ts
class Cipher extends stream.Transform {
  update(data: string, input_encoding: Encoding | undefined, output_encoding: Encoding): string;
}
```

File: lib/test.js (hunk 3 above)
Comment (nguyenhuedang) [reply lib/test.js]: Resolved, thank you

File: lib/key.js (hunk 1 above)
Comment (nguyenhuedang) [reply lib/key.js]: Sure but I think direct usage is still fine

**Pull request merged**